dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorPower.java

```diff
@@ -793,17 +787,15 @@ public List<Event> visitRMWXchg(RMWXchg e) {
     }
 
     @Override
-    public List<Event> visitRMWOp(RMWOp e) {
-        Register resultRegister = e.getResultRegister();
-        IOpBin op = e.getOp();
-        Expression value = e.getMemValue();
+    public List<Event> visitRMWOp(LKMMOpNoReturn e) {
         Expression address = e.getAddress();
         String mo = e.getMo();
 
-        Register dummy = e.getThread().newRegister(resultRegister.getType());
+        Register dummy = e.getThread().newRegister(types.getArchType());
```

Comment:
Are you sure this does not cause problems when we enable BVs?

Reply:
If we enable BVs, then `getArchType` will return a BV type as expected.

`RMWOp` was non-returning: the `resultRegister` did not make any sense to begin with.
In the boogie code, there is no register, so we artificially created one which naturally got the `archType` (what else would we choose?), i.e., the new code produces the exact same result as before (just more directly).